README.md

```diff
@@ -1 +1,114 @@
-# trace-event-logger
\ No newline at end of file
+# Trace Event Logger
+
+This logger is based on JUL to allow fast JSON traces to be written to disk. It is not lockless or nanosecond precise, but is fast and simple to use and configure.
+
+A Snapshot taker is provided in order to write slow transactions onto disk.
+
+An asynchronous writer is available for putting the IO bottleneck on another thread. It is typically much faster than the FileHandler from JUL.
+
+This is a logger helper designed to facilitate entry-exit analysis. The events generated by this logger are saved in a JSON-like message format, providing extra information associated with each event. The typical event types include:
+
+- **Durations:**
+  - **B:** Begin
+  - **E:** End
+  - **X:** Complete (event with a duration field)
+  - **i:** Instant / Info
+
+- **Asynchronous Nested Messages:**
+  - **b:** Nested Begin
+  - **n:** Nested Info
+  - **e:** Nested End
+
+- **Flows:**
+  - **s:** Flow Begin
+  - **t:** Flow Step (Info)
+  - **f:** Flow End
+
+- **Object Tracking:**
+  - **N:** Object Created
+  - **D:** Object Destroyed
+
+- **Mark Events:**
+  - **R:** Marker Event
+
+- **Counter Events:**
+  - **C:** Counter Event
+
+## Usage
+
+To use **durations** and/or **flows**, refer to ScopeLog and FlowScopeLog. Durations are typically used to instrument simple methods, while flows are preferred when there are links to be made with other threads.
+
+To use **Asynchronous Nested Messages**, check traceAsyncStart and traceAsyncEnd].
+
+To use **Object Tracking**, see traceObjectCreation and traceObjectDestruction.
+
+To use free form tracing, check traceInstant.
+
+### Examples
+
+
+To instrument
+
+    public class SlappyWag {
+        public static void main(String[] args) {
+            System.out.println("The program will write hello 10x between two scope logs\n");
+            try (FileWriter fw = new FileWriter("test.txt")) {
+                for (int i = 0; i < 10; i++) {
+                    fw.write("Hello world "+ i);
+                }
+            }
+        }
+    }
+
+one needs to add to the try-with-resources block a scopewriter
+
+    public class SlappyWag {
+        
+        Logger logger = Logger.getAnonymousLogger();
+        
+        public static void main(String[] args) {
+            System.out.println("The program will write hello 10x between two scope logs\n");
+            try (LogUtils.ScopeLog sl = new LogUtils.ScopeLog(logger, Level.FINE, "writing to file"); FileWriter fw = new FileWriter("test.txt")) {
+                for (int i = 0; i < 10; i++) {
+                    fw.write("Hello world "+ i);
+                }
+            }
+        }
+    }
+
+Will generate the following trace
+
+    {"ts":0,"ph":"B","tid":1,"name":"writing to file"}
```

Comment:
would look nice in a `json` block:

\`\`\`json
\<code\>
\`\`\`

Reply:
you probably get the idea for the couple of following java/json code blocks

Reply:
fixed

Reply:
Looks good, thanks! There is a strange glitch in the rendering that I cannot explain, but probably this is still better than no JSON syntax highlight? It might be an artifact of rendering a file from a review and not be present once merged:
![image](https://github.com/eclipse-tracecompass/trace-event-logger/assets/25749063/bc2bbc36-61e5-4c39-9d6d-0419f5beaabf)